src/tools/cephfs/JournalScanner.cc

```diff
@@ -196,6 +197,7 @@ int JournalScanner::scan_events()
       objects_missing.push_back(obj_offset);
       gap = true;
       gap_start = read_offset;
+      gap_len += (object_size - offset_in_obj);
```

Comment:
why not adjust ''read_offset" here?

Reply:
gap_start records the first missing object's offset，here may have 2 or more continuous missing objects,  we must add "if (!gap)" before line 199  if adjust read_offset here

Reply:
> why not adjust ''read_offset" here?

gap_start records the first missing object's offset，here may have 2 or more continuous missing objects, we must add "if (!gap)" before line 199 if adjust read_offset here

Reply:
> 5、delete first two objects 200.0000013e 200.0000013f

In addition to this (manual deletion), are there other conditions that will cause the object to be lost?

Reply:
how about
```
if (!gap) {
   gap_start = read_offset;
   gap = true;
}
read_offset  += object_size - offset_in_obj;
```

Reply:
> how about
> 
> ```
> if (!gap) {
>    gap_start = read_offset;
>    gap = true;
> }
> read_offset  += object_size - offset_in_obj;
> ```

this is ok !     I agree

Reply:
> > 5、delete first two objects 200.0000013e 200.0000013f
> 
> In addition to this (manual deletion), are there other conditions that will cause the object to be lost?

I am sorry!  For me, so far, this situation has not been found! It may be found in the future，sorry, i not sure!

Reply:
> how about
> 
> ```
> if (!gap) {
>    gap_start = read_offset;
>    gap = true;
> }
> read_offset  += object_size - offset_in_obj;
> ```

thanks for your suggestion! I have pushed another commit , pls review again.

Reply:
> > 5、delete first two objects 200.0000013e 200.0000013f
> 
> In addition to this (manual deletion), are there other conditions that will cause the object to be lost?

I think，A suitable condition is osd damaged and  replicas do not meet the minimum number .    line 185 int r = io.read(oid, this_object, INT_MAX, offset_in_obj) ，oid object is considered lost if r < 0,  return val of reading object from rados  < 0 means we cannot read oid object from rados.